pandas/io/tests/test_pickle.py

```diff
@@ -284,6 +284,46 @@ def test_pickle_v0_15_2(self):
         #
         tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
+    def compression_explicit(self, compression):
+        # issue 11666
+        if compression == 'xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path, compression=compression)
+            df2 = pd.read_pickle(path, compression=compression)
+            tm.assert_frame_equal(df, df2)
+
+    def test_compression_explicit(self):
+        compressions = [None, 'gzip', 'bz2', 'xz']
```

Comment:
> Actually in most cases I don't think it's a good design to compress one pickle file into zip file and then enforce there is only one file in that zip.

Agreed, but _get_handle supports reading the single file zip... so you're going to have this feature anyways. I personally would not use zip to compress a single file, but have received datasets that use this method and so it's handy to be able to read them.

Currently, [`to_csv`](http://pandas.pydata.org/pandas-docs/version/0.19.2/generated/pandas.DataFrame.to_csv.html) supports only ‘gzip’, ‘bz2’, ‘xz’ compression. I like this approach -- support reading, but do not make it easy to write.

So I'm not terribly concerned if you don't test for reading the zipped pickle, but since it will work, I think people may use it.

I will open an issue and tag you about moving to a more uniform compression API in the longer term.